ruby_on_rails/testing/model_testing.md

```diff
@@ -0,0 +1,132 @@
+### Introduction
+
+The last lesson covered unit testing and now we will move into model testing. Here you will get an idea of how to test within Rails for model testing as well as a deeper dive of how to structure these tests.
+
+### Lesson Overview
+
+This section contains a general overview of topics that you will learn in this lesson.
+
+*   What are model tests?
+*   How do I run a model test?
+*   How to structure model testing?
+
+## What are model tests?
+
+A model test is a unit test in the context of Ruby on Rails. Model tests in this case are designed to test the business logic of your models. Let's say you need to test a custom validation method for your blog post. In this case you would need to reach into the model aspect of your testing skillset. Before you write the RSpec there will be a few tools you will need to familiarize yourself with.
+
+## Setting the test
+
+The first tool that we are going to use to set up the data is factory bot. Factory bot is a gem that you can use to set up test data and create necessary data for your tests. Adding Factory bot is similar to adding other testing gems. Add the following line to the to the following section in your Gemfile.
+
+~~~
+
+group :development, :test do
+
+  gem "factory_bot_rails"  #Add this line.
+
+end
+
+~~~
+
+Then in your terminal enter `bundle install` to the terminal where it will install the gem. Now we need to move into setting up our test configuration to utilize Factory bot. First were going to navigate to `spec/support/factory_bot.rb` and add the following configuration. 
+
+~~~
+
+RSpec.configure do |config|
+  config.include FactoryBot::Syntax::Methods
+end
+
+~~~
+
+Make sure to require this file in `rails_helper.rb`. We will briefly cover what a factory will look like and some things to keep in mind when creating your data. 
```

Comment:
I think it could be very useful to indicate how to require this file in rails_helper.rb. I am not sure where/how I adopted this, but I believe this is a very common way to make sure to load all support files:
`Dir[Rails.root.join("spec/support/**/*.rb")].each { |f| require f }`